docs/contributing/coding_guide.rst

```diff
@@ -494,56 +494,132 @@ of frustration.
-    make the assumption that the value will be in SI units.
-
-  * A common convention among plasma physicists is to use
-    electron-volts (eV) as a unit of temperature.  Strictly speaking,
-    this unit corresponds not to temperature but is rather a measure
-    of the thermal energy per particle.  Code within PlasmaPy must use
-    the kelvin (K) as the unit of temperature to avoid unnecessary
-    ambiguity.
-
-* PlasmaPy uses the astropy.units package to give physical units to
-  values.
-
-  * All units packages available in Python presently have some
-    limitations, including incompatibility with some NumPy and SciPy
-    functions.  These limitations are due to issues within NumPy
-    itself.  Many of these limitations are being resolved, but require
-    upstream fixes.
-
-* Dimensionless units may be used when appropriate, such as for
-  certain numerical simulations.  The conventions and normalizations
-  should be clearly described in docstrings.
+PlasmaPy uses |astropy.units|_ to assign physical units to values in the
+form of a |Quantity|.
+
+.. code-block:: pycon
+
+   >>> import astropy.units as u
+   >>> 5 * u.m / u.s
+   <Quantity 5. m / s>
+
+Using |astropy.units|_ improves compatibility with Python packages in
+adjacent fields such as astronomy and heliophysics. To get started with
+|astropy.units|_, check out this `example notebook on units`_.
+
+  .. caution::
+
+     Some `scipy` functions silently drop units when used on |Quantity|
+     instances.
+
+* Only SI units should be used within PlasmaPy, unless there is a strong
+  justification to do otherwise. Example notebooks may occasionally use
+  other unit systems to show the flexibility of |astropy.units|_.
+
+* Use operations between |Quantity| instances except when needed for
+  performance. To improve performance in |Quantity| operations, check
+  out `performance tips
+  <https://docs.astropy.org/en/stable/units/index.html#performance-tips>`__
+  for |astropy.units|_.
+
+* Use unit annotations with the |validate_quantities| decorator to
+  validate |Quantity| arguments and return values.
+
+  .. code-block:: python
+
+     from plasmapy.utils.decorators.validators import validate_quantities
+
+     @validate_quantities(
+        n={"can_be_negative": False},
+        validations_on_return={"equivalencies": u.dimensionless_angles()},
+     )
+     def inertial_length(n: u.m ** -3, ...) -> u.m:
+         ...
+
+  .. caution::
+
+     Recent versions of Astropy_ allow unit-aware |Quantity|
+     annotations such as ``u.Quantity[u.m]``. However, these annotations
+     are not yet compatible with |validate_quantities|.
+
+* Avoid using electron-volts as a unit of temperature within PlasmaPy
+  because it is defined as a unit of energy. However, functions in
+  `plasmapy.formulary` and elsewhere should accept temperatures in units
+  of electron-volts, which can be done using |validate_quantities|.
```

Comment:
Not to delay this PR, but, random idea... perhaps this should actually be configurable. I can tell you now that for many a fusioneer output in K is going to be practically useless.

Reply:
Brought this up in #1321!